manifests/init.pp

```diff
@@ -56,17 +56,22 @@
 #
 class svckill (
   Array[String]               $ignore       = [],
+  Array[String]               $ignore_defaults = [],
   Array[Stdlib::Absolutepath] $ignore_files = [],
-  Boolean                     $verbose      = true
+  Boolean                     $verbose      = true,
+  Boolean                     $debug        = false
 ){
   include '::svckill::ignore::collector'
-
+  $combined_ignore_list = $ignore + $ignore_defaults
+  if ($debug == true) {
```

Comment:
I can see why you handled the debugging this way, but it's a bit weird: 
*  There are now two class parameters for different types of verbosity 
* `$debug` logs its output using a different mechanism than `$verbose`.
* `$debug`'s method of logging adds resources to the catalog 

**A suggestion:** Handle this reporting in the `svckill` type's `change_to_s` method (the same place `$verbose` is handled) and log it using `Puppet.debug` (which reports output only when you run in `--debug` mode).  

* Both lists would be logged using the same mechanism
* You won't need to add the extra `$debug` parameter just to add a new tier of verbosity
* Logging won't affect the catalog.